### ISSUE TITLE
False negatives for consider-using-min-builtin/consider-using-max-builtin

### ISSUE THREAD

**Issue by EugeneZelenko** [pylint-dev/pylint#8947]
### Bug description

`consider-using-min-builtin/consider-using-max-builtin` did not catch next situations:

```
value = 5
max_value = 10

if value > max_value:
    max_value = value

if value > float(max_value):
    value = float(max_value)
```

### Command used

```shell
pylint --load-plugins=pylint.extensions.code_style --enable=all test.py
```

### Pylint output

```shell
************* Module pylint_test
pylint_test.py:1:0: C0114: Missing module docstring (missing-module-docstring)
pylint_test.py:1:0: C0103: Constant name "value" doesn't conform to UPPER_CASE naming style (invalid-name)
pylint_test.py:2:0: C0103: Constant name "max_value" doesn't conform to UPPER_CASE naming style (invalid-name)
pylint_test.py:5:4: C0103: Constant name "max_value" doesn't conform to UPPER_CASE naming style (invalid-name)

-------------------------------------------------------------------
Your code has been rated at 3.33/10 (previous run: 10.00/10, -6.67)
```

### Expected behavior

First `if` statement should be suggested to be replaced with `max_value = max(max_value, value)`.

Second - with `value = min(value, float(max_value))`.

### Pylint version

```shell
pylint 2.17.4
astroid 2.15.5
Python 3.11.4 (main, Jun 26 2023, 16:57:25) [GCC 7.5.0]
```

**Comment by EugeneZelenko** [comment]
Same for

```
if value < min_value:
    min_value = value
```

**Comment by Pierre-Sassoulas** [comment]
Thank you for opening the issue, I can reproduce on main.

**Comment by EugeneZelenko** [comment]
Another example found it the wild:

```
offset = 1
if offset + value < min_value:
    min_value = offset + value
```

Could be generalized to arbitrary expression.

**Comment by rhyn0** [comment]
If no one else wants to work on this, could I be assigned it?

**Comment by Pierre-Sassoulas** [comment]
Sure, thank you for working on it !
